### DISCUSSION TITLE
Metadata not showing up

### DISCUSSION THREAD

**Discussion by fcimeson** [photoprism/photoprism#3029] [Q&A]
Hello, I'm trying to import my library into Photoprism and I have some of my data stored in the images metadata. The Description and PersonInImage fields shows up in Photoprism but the other metadata tags supported in this [list](https://photoprism.app/kb/metadata) don't (Notes, CatalogSets).

Please advise.

**Comment by fcimeson** [comment]
PS. I'm trying to use the other fields as a hash for the albums because the search doesn't do exact phrases, e.g., when I search for "1 month old" I get results that also match "2 months old", etc.

**Reply by graciousgrey** [reply]
What exactly are you trying to search for? Have you seen: https://docs.photoprism.app/user-guide/search/filters/ ?

> The Description and PersonInImage fields shows up in Photoprism but the other metadata tags supported in this [list](https://photoprism.app/kb/metadata) don't (Notes, CatalogSets). 

To be able to guess what might be the issue with your metadata - the exif data of one of your images would help `exiftool -j example.jpg` :)

**Reply by fcimeson** [reply]
I'm trying to search for my hashes (added to Notes and CatalogSets).
```
exiftool -j test.jpg
[{
  "SourceFile": "test.jpg",
  "ExifToolVersion": 12.52,
  "FileName": "test.jpg",
  "Directory": ".",
  "FileSize": "1945 kB",
  "FileModifyDate": "2022:12:24 07:16:47-05:00",
  "FileAccessDate": "2022:12:24 07:16:57-05:00",
  "FileInodeChangeDate": "2022:12:24 07:16:47-05:00",
  "FilePermissions": "-rw-r--r--",
  "FileType": "JPEG",
  "FileTypeExtension": "jpg",
  "MIMEType": "image/jpeg",
  "ExifByteOrder": "Little-endian (Intel, II)",
  "Make": "Google",
  "Model": "Pixel 6",
  "Orientation": "Horizontal (normal)",
  "XResolution": 72,
  "YResolution": 72,
  "ResolutionUnit": "inches",
  "Software": "HDR+ 1.0.478604551nd",
  "ModifyDate": "2022:11:28 20:59:21",
  "YCbCrPositioning": "Centered",
  "ExposureTime": "1/8",
  "FNumber": 1.9,
  "ExposureProgram": "Program AE",
  "ISO": 2547,
  "ExifVersion": "0232",
  "DateTimeOriginal": "2022:11:28 20:59:21",
  "CreateDate": "2022:11:28 20:59:21",
  "OffsetTime": "+00:00",
  "OffsetTimeOriginal": "+00:00",
  "OffsetTimeDigitized": "+00:00",
  "ComponentsConfiguration": "Y, Cb, Cr, -",
  "ShutterSpeedValue": "1/8",
  "ApertureValue": 1.9,
  "BrightnessValue": -4.82,
  "ExposureCompensation": 0,
  "MaxApertureValue": 1.9,
  "SubjectDistance": "2.592 m",
  "MeteringMode": "Center-weighted average",
  "Flash": "Off, Did not fire",
  "FocalLength": "6.8 mm",
  "SubSecTime": 280,
  "SubSecTimeOriginal": 280,
  "SubSecTimeDigitized": 280,
  "FlashpixVersion": "0100",
  "ColorSpace": "sRGB",
  "ExifImageWidth": 4080,
  "ExifImageHeight": 3072,
  "InteropIndex": "R98 - DCF basic file (sRGB)",
  "InteropVersion": "0100",
  "SensingMethod": "One-chip color area",
  "SceneType": "Directly photographed",
  "CustomRendered": "Custom",
  "ExposureMode": "Auto",
  "WhiteBalance": "Auto",
  "DigitalZoomRatio": 0,
  "FocalLengthIn35mmFormat": "24 mm",
  "SceneCaptureType": "Standard",
  "Contrast": "Normal",
  "Saturation": "Normal",
  "Sharpness": "Normal",
  "SubjectDistanceRange": "Close",
  "LensMake": "Google",
  "LensModel": "Pixel 6 back camera 6.81mm f/1.85",
  "CompositeImage": "Composite Image Captured While Shooting",
  "GPSLatitudeRef": "North",
  "GPSLongitudeRef": "West",
  "GPSAltitudeRef": "Above Sea Level",
  "GPSTimeStamp": "20:59:19",
  "GPSImgDirectionRef": "Magnetic North",
  "GPSImgDirection": 150,
  "GPSDateStamp": "2022:11:28",
  "Compression": "JPEG (old-style)",
  "ThumbnailOffset": 1343,
  "ThumbnailLength": 15381,
  "CurrentIPTCDigest": "e34b05b0282c7fc003351db3115910ba",
  "CatalogSets": "31204d6f6e7468204f6c64,30204d6f6e746873204f6c64",
  "ApplicationRecordVersion": 4,
  "XMPToolkit": "Image::ExifTool 12.52",
  "HasExtendedXMP": "40D3708DE4889E1260DCED896F91775D",
  "Notes": "0xfa888fdd,0xf280e05a",
  "Description": "0 Months Old,1 Month Old",
  "PersonInImage": "Edward Imeson",
  "ShotLogData": "(Binary data 579 bytes, use -b option to extract)",
  "JFIFVersion": 1.02,
  "ProfileCMMType": "",
  "ProfileVersion": "4.0.0",
  "ProfileClass": "Display Device Profile",
  "ColorSpaceData": "RGB ",
  "ProfileConnectionSpace": "XYZ ",
  "ProfileDateTime": "2016:12:08 09:38:28",
  "ProfileFileSignature": "acsp",
  "PrimaryPlatform": "Unknown ()",
  "CMMFlags": "Not Embedded, Independent",
  "DeviceManufacturer": "Google",
  "DeviceModel": "",
  "DeviceAttributes": "Reflective, Glossy, Positive, Color",
  "RenderingIntent": "Perceptual",
  "ConnectionSpaceIlluminant": "0.9642 1 0.82491",
  "ProfileCreator": "Google",
  "ProfileID": "75e1a6b13c34376310c8ab660632a28a",
  "ProfileDescription": "sRGB IEC61966-2.1",
  "ProfileCopyright": "Copyright (c) 2016 Google Inc.",
  "MediaWhitePoint": "0.95045 1 1.08905",
  "MediaBlackPoint": "0 0 0",
  "RedMatrixColumn": "0.43604 0.22249 0.01392",
  "GreenMatrixColumn": "0.38512 0.7169 0.09706",
  "BlueMatrixColumn": "0.14305 0.06061 0.71391",
  "RedTRC": "(Binary data 32 bytes, use -b option to extract)",
  "ChromaticAdaptation": "1.04788 0.02292 -0.05019 0.02959 0.99048 -0.01704 -0.00922 0.01508 0.75168",
  "BlueTRC": "(Binary data 32 bytes, use -b option to extract)",
  "GreenTRC": "(Binary data 32 bytes, use -b option to extract)",
  "ImageWidth": 4080,
  "ImageHeight": 3072,
  "EncodingProcess": "Baseline DCT, Huffman coding",
  "BitsPerSample": 8,
  "ColorComponents": 3,
  "YCbCrSubSampling": "YCbCr4:2:0 (2 2)",
  "HDRPMakerNote": "(Binary data 68512 bytes, use -b option to extract)",
  "Aperture": 1.9,
  "ImageSize": "4080x3072",
  "Megapixels": 12.5,
  "ScaleFactor35efl": 3.5,
  "ShutterSpeed": "1/8",
  "SubSecCreateDate": "2022:11:28 20:59:21.280+00:00",
  "SubSecDateTimeOriginal": "2022:11:28 20:59:21.280+00:00",
  "SubSecModifyDate": "2022:11:28 20:59:21.280+00:00",
  "ThumbnailImage": "(Binary data 15381 bytes, use -b option to extract)",
  "GPSAltitude": "100.8 m Above Sea Level",
  "GPSDateTime": "2022:11:28 20:59:19Z",
  "GPSLatitude": "64 deg 34' 51.84\" N",
  "GPSLongitude": "21 deg 40' 18.79\" W",
  "CircleOfConfusion": "0.009 mm",
  "DOF": "20.08 m (1.38 - 21.46 m)",
  "FOV": "73.7 deg",
  "FocalLength35efl": "6.8 mm (35 mm equivalent: 24.0 mm)",
  "GPSPosition": "64 deg 34' 51.84\" N, 21 deg 40' 18.79\" W",
  "HyperfocalDistance": "2.94 m",
  "LightValue": 0.2,
  "LensID": "Pixel 6 back camera 6.81mm f/1.85"
}]
```

And yes I have seen that page.

**Reply by lastzero** [reply]
Using the general text search? It only indexes human readable words as not to clutter the index with random strings (there are a lot in metadata). Hashes or other random / binary data can be searched only if you use a filter for a specific field, provided it exists.

**Comment by fcimeson** [comment]
Notes and Catalogue Sets isn't listed as a [searchable tag](https://docs.photoprism.app/user-guide/search/filters/#filter-reference). I guess that is what I'm getting at, is what metadata can I populate in the image that I can search?

The documentation is not clear as to what metadata is used from the image file itself and what metadata is used from photoprism - I'm trying to understand what metadata from the file I can use. For example, my files use the PersonInImage tags which seem to fall into the search category of subject but when I try to search for a specific value (e.g., subject:"test") it matches other criteria as well (e.g., the ML identified faces, not my PersonInImage tag).

Hence, why I'm trying to use my own hash.

**Reply by lastzero** [reply]
I'm celebrating Christmas with my family right now, so I can't check the code & docs to send references. Generally speaking, you can search those fields, for which filters exist. Plus there is a general keyword search including, yes, keywords, and keywords extracted from the Title, Description and Subject. Excluding internal notes, unique id's, camera lens, camera model, license description, video codec, and any other generic technical metadata that would fall into such category. We welcome pull requests for our docs to improve them if needed, which seems to be the case based on your question.

**Reply by fcimeson** [reply]
Ok, well for sure enjoy your Christmas and thank you so much for the support. I look forward to your reply when you have time.

**Reply by lastzero** [reply]
You can't work with that? I was hoping you can now search the docs (and public code) on your own, to figure out further details if needed, now that I've explained the general concept 😉

**Reply by graciousgrey** [reply]
[This list](https://photoprism.app/kb/metadata) shows what metadata from your image is read into which PhotoPrism field.
E.g. The CatalogSet field from your image will be imported into the subject field in PhotoPrism. Whereas the comment EXIF field from your image will be imported as notes PhotoPrism.
[This list](https://docs.photoprism.app/user-guide/search/filters/#filter-reference) shows for which PhotoPrism field a search filter exists. 
There is no search filter for subject and notes yet. The existing subject search filter is an alias for the person filter it does not relate to the subject field on the details tab - I guess we should improve the naming here. 
As far as I remember we do already have feature requests to add search filters for the details subject.
I hope its a bit more clear now :)

**Reply by fcimeson** [reply]
Thanks, I have been rigorously checking those lists but there have been a few gotchas.
- The internal subject field will only take on the value of one of the values from your image file. For example, if you have CatalogSets and PersonInImage, then it only takes on the PersonInImage value.
- There isn't a search filter for the Description field, which is why I'm trying to use hashes.
- The search filter for subject doesn't seem to work for my application, e.g., I tried subject:"Album-0001" and it didn't work but just searching for "Album-0001" did work. Based on the filter notes, maybe it only works for people???
- I'm going to try and use the notes/comment filed but there is no search filter for notes so I'm not sure what I'm in for.
- There was mention of hashes not being searchable (or maybe they don't even import?). It's not clear to me what the nuances of this are yet but I'm providing the prefix of "Album-" to my hash to hopefully avoid this headache.

**Reply by lastzero** [reply]
Again, notes are private notes, your personal comments like what you want to edit. Thus there is no search filter and it's so far not included in search.

**Reply by fcimeson** [reply]
Ok, so what can I use (subject is already used by PersonInImage and photoprism isn't a union of the tags, it just picks one of them)?

**Reply by fcimeson** [reply]
Also, can you provide context as to how or how not tag values are filtered. You previously stated that hash tags are filtered.

**Reply by lastzero** [reply]
Whatever comes first. If your main use case is to store and find a unique ID, the instance / document ID are well suited as they are indexed and not just arbitrary text fields.

**Reply by fcimeson** [reply]
The filter page doesn't reference instance/document id. Are they searchable?

**Reply by lastzero** [reply]
Hashes are not filtered, they are just not indexed as part of the full TEXT search (as Text consists of Words, not random characters). You could use one of the fields designed for unique IDs or do a feature request that describes your exact use case and optionally one or more proposed solutions. To get to the bottom of technical details, if you interested, it seems best to take a look at our public source code.

**Reply by lastzero** [reply]
Yes they should be. On my phone now. If not, we can make it so.

**Reply by fcimeson** [reply]
Awesome, I will give it a try. Thanks again for all of the support.

**Comment by fcimeson** [comment]
Unfortunately, I'm not able to search my tags (document ID). Should I fill out a user story?

Update: Keywords gets the job done. Not my first choice but it works.

**Reply by lastzero** [reply]
Implemented:
- #3035

Can be tested with the next development preview build, which will be available soon.
